learning_journal/learning_journal/tests.py

```diff
@@ -0,0 +1,161 @@
+def setup_test_env():
+    os.environ["DATABASE_URL"] = 'postgres://VMB@localhost:5432/mockdb_ljtesting'
+
+@pytest.fixture(scope="function")
+def sqlengine(request):
+    """Set up sql engine."""
+    config = testing.setUp(settings=DB_SETTINGS)
+    config.include(".models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    """Set up new new session."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture(scope="function")
+def populated_db(request, sqlengine):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    with transaction.manager:
+        session.add(MyEntry(title="Test Title", body="This is a test entry, James is being awesome.", creation_date=datetime.datetime.utcnow()))
+
+    def teardown():
+        with transaction.manager:
+            session.query(MyEntry).delete()
+
+    request.addfinalizer(teardown)
+
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """Call a dummy request."""
+    return testing.DummyRequest(dbsession=new_session)
+
+
+def test_home_view(dummy_request, new_session):
+    """Test entries are in home view."""
+    from .views.default import home_view
+    new_session.add(MyEntry(title="test", body="this is a test", creation_date=datetime.datetime.utcnow()))
+    new_session.flush()
+    info = home_view(dummy_request)
+    assert "entries" in info
+
+
+def test_detail_view(new_session):
+    """Test detail view has a body."""
+    from .views.default import detail_view
+    request = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyEntry(title="test", body="this is a test", creation_date=datetime.datetime.utcnow()))
+    new_session.flush()
+    request.matchdict = {'id': '1'}
+    info = detail_view(request)
+    assert "this is a test" in info['entry'].body
+
+
+def test_create_view():
+    """Test create view."""
+    from .views.default import create_view
+    request = testing.DummyRequest()
+    create_view(request)
+    assert request.response.status_code == 200
+
+
+def test_edit_view(new_session):
+    """Test update view has a body."""
+    from .views.default import edit_view
+    request = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyEntry(title="test", body="this is a test", creation_date=datetime.datetime.utcnow()))
+    new_session.flush()
+    request.matchdict = {'id': '1'}
+    info = edit_view(request)
+    assert info["entry"].body == "this is a test"
```

Comment:
What happens for a post request?  When you use an ID that isn't in the database? When you post to an ID that isn't in the database?